src/duct/__main__.py

```diff
@@ -91,6 +91,71 @@ def max(self, other: ProcessStats) -> ProcessStats:
         )
 
 
+@dataclass
+class Suffix:
+    stdout: str = "stdout"
+    stderr: str = "stderr"
+    usage: str = "usage.json"
+    info: str = "info.json"
+
+    def __iter__(self) -> Generator:
+        for each in fields(self):
+            yield each.name, getattr(self, each.name)
+
+
+@dataclass
+class LogPaths:
+    stdout: str
+    stderr: str
+    usage: str
+    info: str
+
+    def __iter__(self) -> Generator:
+        for each in fields(self):
+            yield each.name, getattr(self, each.name)
+
+    def __post_init__(self) -> None:
+        self._prefix = ""
+
+    @property
+    def prefix(self) -> str:
+        return self._prefix  # type: ignore
+
+    @classmethod
+    def create(cls, output_prefix: str, pid: None | int = None) -> LogPaths:
+        datetime_filesafe = datetime.now().strftime("%Y.%m.%dT%H.%M.%S")
+        formatted_prefix = output_prefix.format(
+            pid=pid, datetime_filesafe=datetime_filesafe
+        )
+        args_dict = {name: f"{formatted_prefix}{suffix}" for name, suffix in Suffix()}
+        new = cls(**args_dict)
+        new._prefix = formatted_prefix
```

Comment:
Why not just make `_prefix` a normal attribute of `LogPaths` that is initialized from a constructor argument instead of doing this stupid dance?  If your goal is just to keep `prefix` out of `self.__iter__()`, you can filter it out in that method.